### ISSUE TITLE
[BUG] Image handling inconsistency between "Harvester Terraform harvester_image data source" vs. "UI created Image"

### ISSUE THREAD

**Issue by Martin-Weiss** [harvester/harvester#2443]
**Describe the bug**
When creating a resource harvester_virtualmachine we have to specify the image with its "id / name".
When creating an image with the harvester UI - we can only specify the "display name" for the image and the "name" of the image is created with "generateName: image-".

Now when we want to use the data source data.harvester_image.image.id like documented here https://github.com/harvester/terraform-provider-harvester/blob/master/examples/data-sources/harvester_image/data-source.tf - we can not refer to the display name that had been used via creation of the image via UI.

In case we create the image via the harvester_image resource - the name can be specified and then can be used - but we have images pre-created via UI.

**To Reproduce**
See above.

**Expected behavior**
The way how images are created via UI and via terraform need to match.

And in case the UI gives a random name to an image - the terraform data source needs to be able to get the object via "display name" instead of "name", too.

**Environment**
 - Harvester ISO version: 1.0.2

**Comment by futuretea** [comment]
we can use [external data source](https://registry.terraform.io/providers/hashicorp/external/latest/docs/data-sources/data_source) to get the generated name
---
1. install `jq` `kubectl`
2. create a shell script `get_image_generated_name.sh` with the following contents:
```bash
#!/bin/bash
set -e
eval "$(jq -r '@sh "DISPLAY_NAME=\(.display_name) NAMESPACE=\(.namespace)"')"
NAME=$(kubectl -n ${NAMESPACE} get vmimage -o json | jq -r --arg display_name "$DISPLAY_NAME" '.items[] | select(.spec.displayName == $display_name) | .metadata.name')
jq -n --arg name "$NAME" '{"name":$name}'
```

3. add external provider to provider.tf:
```hcl
terraform {
  required_providers {
    harvester = {
      source = "harvester/harvester"
      version = "0.4.0"
    }
    external = {
      source = "hashicorp/external"
      version = "2.2.2"
    }
  }
}
provider "harvester" {
}
provider "external" {
}
```

4. use external data source
```hcl
data "external" "get_ubuntu1804_image_name" {
  program = ["bash","get_image_generated_name.sh"]
  query = {
    display_name = "ubuntu-18.04-minimal-cloudimg-amd64.img"
    namespace = "default"
  }
}

data "harvester_image" "ubuntu1804" {
  name      = data.external.get_ubuntu1804_image_name.result.name
  namespace = "default"
}
```

```bash
terraform apply
```

**Comment by Martin-Weiss** [comment]
Thanks for the workaround!

Do we have any chance to get this problem fixed in the Harvester UI (allowing image Name to be specified in addition to the Display Name) or could we get the data source in the Harvester terraform provider fixed so that it also supports “display name” directly?

**Comment by futuretea** [comment]
I will try to enhance terraform provider to support “display _name”

**Comment by TachunLin** [comment]
Verified **fixed** on harvester-terraform-provider `0.5.1` with Harvester `v1.0.3`. Close this issue.

### Result
Now we can use the data source refer to the display name that had been used via creation of the image via UI.

* Image resource created in image UI
![image](https://user-images.githubusercontent.com/29251855/185184768-18708aaf-96a4-40c2-8d8f-247a21d42705.png)

* Image state of running `terraform show`
```
davidtclin@localhost:~/Documents/Harvester/workspace/terraform-provider-harvester/examples/test> terraform show
# data.harvester_image.ubuntu1804:
data "harvester_image" "ubuntu1804" {
    display_name       = "ubuntu-18.04-minimal-cloudimg-amd64.img"
    id                 = "harvester-public/image-7w8tp"
    name               = "image-7w8tp"
    namespace          = "harvester-public"
    progress           = 100
    size               = 202637312
    source_type        = "download"
    state              = "Active"
    storage_class_name = "longhorn-image-7w8tp"
    tags               = {}
    url                = "https://cloud-images.ubuntu.com/minimal/releases/bionic/release/ubuntu-18.04-minimal-cloudimg-amd64.img"
}
...
```

### Test Information
* Test Environment: 1 node harvester on local kvm machine
* Harvester version: 0.5.1

### Verify Steps
1. Download latest terraform-provider [terraform-provider-harvester_0.5.1_linux_amd64.zip](https://github.com/harvester/terraform-provider-harvester/releases/download/v0.5.1/terraform-provider-harvester_0.5.1_linux_amd64.zip)
2. Extra the zip file
3. Create the install-terraform-provider-harvester.sh with the following content
```
#!/usr/bin/env bash
[[ -n $DEBUG ]] && set -x
set -eou pipefail

usage() {
    cat <<HELP
USAGE:
    install-terraform-provider-harvester.sh
HELP
}

version=0.5.1
arch=linux_amd64
terraform_harvester_provider_bin=./terraform-provider-harvester

terraform_harvester_provider_dir="${HOME}/.terraform.d/plugins/registry.terraform.io/harvester/harvester/${version}/${arch}/"
mkdir -p "${terraform_harvester_provider_dir}"
cp ${terraform_harvester_provider_bin} "${terraform_harvester_provider_dir}/terraform-provider-harvester_v${version}"
```
4. Rename the extraced `terraform-provider-harvester_v0.5.1` to `terraform-provider-harvester` in the same folder with the install script
5. Execute the installation script 
6. Clone the terraform-provider project 
7. Get the Harvester kubeconfig and place to the specific config file 
8. Specify the kubeconfig path in `provider.tf`
  ```
  terraform {
    required_version = ">= 0.13"
    required_providers {
      harvester = {
        source  = "harvester/harvester"
        version = "0.5.1"
      }
    }
  }
  
  provider "harvester" {
    kubeconfig = "/home/davidtclin/.kube/config"
  }
  ```  
9. Create a image ubuntu-18.04-minimal-cloudimg-amd64.img in the namespace harvester-public form UI side
10. Create a data source harvester_image.ubuntu1804 in resource.tf 
```
data "harvester_image" "ubuntu1804" {
  namespace    = "harvester-public"
  display_name = "ubuntu-18.04-minimal-cloudimg-amd64.img"
}
```
11. Execute `terraform init -upgrade`
12. Execute `terraform apply`

13. Execute `terraform show`
```
...
...
Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

harvester_ssh_key.davidtclin: Creating...
harvester_ssh_key.davidtclin: Creation complete after 0s [id=default/davidtclin]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

* Execute `terraform show`
```
davidtclin@localhost:~/Documents/Harvester/workspace/terraform-provider-harvester/examples/test> terraform show
# data.harvester_image.ubuntu1804:
data "harvester_image" "ubuntu1804" {
    display_name       = "ubuntu-18.04-minimal-cloudimg-amd64.img"
    id                 = "harvester-public/image-7w8tp"
    name               = "image-7w8tp"
    namespace          = "harvester-public"
    progress           = 100
    size               = 202637312
    source_type        = "download"
    state              = "Active"
    storage_class_name = "longhorn-image-7w8tp"
    tags               = {}
    url                = "https://cloud-images.ubuntu.com/minimal/releases/bionic/release/ubuntu-18.04-minimal-cloudimg-amd64.img"
}
...
```
